### PR TITLE
Replace destats with stats

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

**Comment by oscardssmith** [comment]
this change is at least somewhat breaking.

File: src/destats.jl [hunk 1]
```diff
@@ -24,7 +24,7 @@ Statistics from the differential equation solver about the solution process.
 - maxeig: Maximum eigenvalue over the solution. This is only computed if the
   method is an auto-switching algorithm.
 """
-mutable struct DEStats
+mutable struct Stats
```
Comment (ChrisRackauckas) [inline src/destats.jl:27]: missing a deprecation `const`

Comment (avik-pal) [reply src/destats.jl:27]: Do we keep the names for internal structs the same https://github.com/SciML/DelayDiffEq.jl/pull/258#discussion_r1122316765? Or change them

**Comment by avik-pal** [comment]
Only thing that fails is ForwardDiff Dual detection with https://github.com/SciML/SciMLBase.jl/pull/391

File: Project.toml [hunk 2]
```diff
@@ -73,7 +73,7 @@ PreallocationTools = "0.4"
 RecursiveArrayTools = "2"
 Reexport = "1.0"
 Requires = "1.0"
-SciMLBase = "1.84"
+SciMLBase = "1.89"
```
Comment (ChrisRackauckas) [inline Project.toml:76]:
```suggestion
SciMLBase = "1.90"
```

**Pull request merged**